src/main/java/sirius/biz/analytics/charts/explorer/ChartFactory.java

```diff
@@ -0,0 +1,265 @@
+     * @param start            the start of the selected period
+     * @param end              the end of the selected period
+     * @param granularity      the granularity to provide
+     * @param comparisonPeriod the selected type of comparison period to provide
+     * @param output           the output to write all JSON to
+     * @throws Exception in case of any error when computing the chart
+     */
+    protected void computeData(@Nullable O object,
+                               LocalDate start,
+                               LocalDate end,
+                               Granularity granularity,
+                               ComparisonPeriod comparisonPeriod,
+                               JSONStructuredOutput output) throws Exception {
+        List<String> hints = new ArrayList<>();
+        output.property(OUTPUT_LABEL, getChartLabel(object));
+        output.property(OUTPUT_SUB_LABEL, getChartSubLabel(object));
+        output.property(OUTPUT_DESCRIPTION, getChartDescription(object));
+        outputReferences(object, output);
+
+        computeData(object, start, end, granularity, comparisonPeriod, hints::add, output);
+
+        output.array(OUTPUT_HINTS, OUTPUT_HINT, hints);
+    }
+
+    /**
+     * Outputs all visible references.
+     *
+     * @param object the selected reference object
+     * @param output the output to write all JSON to
+     */
+    private void outputReferences(O object, JSONStructuredOutput output) {
+        output.beginArray(OUTPUT_REFERENCES);
+        for (ChartFactory<O> referencedProvider : referencedFactories()) {
+            output.beginObject(OUTPUT_REFERENCE);
+            output.property(OUTPUT_LABEL, referencedProvider.getLabel());
+            if (referencedProvider.resolver() != null) {
+                output.property(OUTPUT_IDENTIFIER,
+                                referencedProvider.getName() + ":" + referencedProvider.resolver()
+                                                                                       .fetchIdentifier(object));
+            } else {
+                output.property(OUTPUT_IDENTIFIER, referencedProvider.getName());
+            }
+            output.endObject();
+        }
+        output.endArray();
+    }
+
+    /**
+     * Computes the actual chart data and outputs it into the given output.
+     *
+     * @param object           the selected reference object
+     * @param start            the start of the selected period
+     * @param end              the end of the selected period
+     * @param granularity      the granularity to provide
+     * @param comparisonPeriod the selected type of comparison period to provide
+     * @param hints            a consumer which can be provided with hints to show for a chart
+     * @param output           the output to write all JSON to
+     * @throws Exception in case of any error when computing the chart
+     */
+    protected abstract void computeData(O object,
```

Comment:
The name of these functions don't really point to which should be overwritten by subclasses as they are essentially the same (even the parameter list looks nearly the same). Could we differentiate them a bit/